### PR TITLE
DockerTarget: Warn users about container removal.

### PR DESCRIPTION
**Pull request by ltfish**
OMG I spent 24 hours on an issue caused by this...

### CODE REVIEW

**Comment by ltfish** [comment]
More context: `DockerImageTarget.remove()` is called when `DockerImageTarget.__del__()` is triggered. To be honest, I feel calling `remove()` and removing a container when the variable goes out of scope is _way_ too implicit and is extremely annoying to debug. Since explicit is always better than implicit in Python, I think a better fix is to not call `remove()` when `__del__()` is called.

Also, afaik `__del__()` is not guaranteed to be called in Python 3. So the logic is wrong if we rely on `__del__()` to clean up Docker containers at exit.

**Comment by zardus** [comment]
On further thought, the problem is that this makes a DockerTarget inherently different than a LocalTarget, in that for the former, you have to call `remove` and for the former, you cannot. Can you make this a more general thing, so we can update all the testcases to match?

**Comment by ltfish** [comment]
Why do you want to remove a local target?

A docker image target generates a docker container as a “temporary file” and should be removed at `__exit__`, which is the same as a local target removing all temporary files that it generates at `__exit__`. If the user chooses to use the DockerImageTarget without context manager, they should be responsible for managing its life cycle, which should be the same if they use the local target.

In conclusion, I don’t see how the two targets have different behaviors.

**Comment by zardus** [comment]
You've convinced me.
